src/03-ticket-calculator.js

```diff
@@ -109,10 +141,57 @@ function calculateTicketPrice(ticketData, ticketInfo) {}
     purchaseTickets(tickets, purchases);
     //> "Ticket type 'discount' cannot be found."
  */
-function purchaseTickets(ticketData, purchases) {}
+function purchaseTickets(ticketData, purchases) {
+  
+ let finaltTotal = 0;
+ let tempTotal = [];
+ let finalStr = 'Thank you for visiting the Dinosaur Museum!\n-------------------------------------------'
+ let j = 0;
+ let tempStr = ""
+  for (const purchase of purchases) {
+  // Errors msgs 
+    if (!['general','membership'].includes(purchase.ticketType)) {
+      return "Ticket type 'incorrect-type' cannot be found."
+      }
+    if (!['child','adult','senior'].includes(purchase.entrantType)) {
+      return "Entrant type 'incorrect-entrant' cannot be found."
+      }
+    
+    for (const extra of purchase.extras) {
+      if (!['movie','education','terrace'].includes(extra)) {
+        return "Extra type 'incorrect-extra' cannot be found."
+        }
+    } 
```

Comment:
Same note as above here. Your returns can be more dynamic. Also another approach you could have taken was to leverage the `calculateTicketPrice` function. If the return is a string then your know it is an error message and return that, if not you would add the price for it ticket to a total .